### DISCUSSION TITLE
UDP Stream from gNB to UE crashes (when using antennas)

### DISCUSSION THREAD

**Discussion by oliverxsch** [srsran/srsRAN_Project#347] [General Help]
Hi,
i started experimenting with srsRAN and i tried streaming via VLC (UDP).
It works for a few seconds but then the connection crashes.
This only happens when i use antennas though. When im using cables with 30db attenuators (with the same config) it works most of the time.
It tried tweeking some values but i was not able to make the connection stable.
I attached the logs and console traces.

First PC:
Specs: i9 13900, 64GB Ram
OS: Ubuntu 22.04.03 LTS
UHD 3.15 LTS
Newest srsRAN Project and newest Open5Gs.
Connected via a 10G-Link to a USRP X310
"gNB"

Second PC:
Specs: i9 13900, 64GB Ram
OS: Ubuntu 22.04.03 LTS
UHD 3.15 LTS
Newest srsRAN-4G Project.
Connected via a 10G-Link to another USRP X310
"UE"

Both USRPs are connected to an CDA 2990 G for their sync and clock.

Thanks in advance for your help.

[ue gnb udp stream.zip](https://github.com/srsran/srsRAN_Project/files/13528292/ue.gnb.udp.stream.zip)

**Comment by pgawlowicz** [comment]
could you try the following change for the UE config:

```
device_args = type=x300,addr=192.168.43.2,master_clock_rate=184.32e6,send_frame_size=8000,recv_frame_size=8000,clock_source=external,sync=external
```

ie `clock` --> `clock_source`
and post the console output here

**Reply by oliverxsch** [reply]
Just did that and if i change that, it does not connect anymore. If i revert the change it does connect again.
UE Console:
inets@inets-LAB:~/srsRAN_4G/build$ sudo srsue ue_x300.conf
Active RF plugins: libsrsran_rf_uhd.so
Inactive RF plugins: 
Reading configuration file ue_x300.conf...

Built in Release mode using commit eea87b1d8 on branch master.

Opening 1 channels in RF device=uhd with args=type=x300,addr=192.168.43.2,master_clock_rate=184.32e6,send_frame_size=8000,recv_frame_size=8000,clock_source=external,sync=external
Supported RF device list: UHD file
[INFO] [UHD] linux; GNU C++ version 11.4.0; Boost_107400; UHD_3.15.0.0-74-ge35f66e8
[INFO] [LOGGING] Fastpath logging disabled at runtime.
Opening USRP channels=1, args: type=x300,addr=192.168.43.2,master_clock_rate=184.32e6,send_frame_size=8000,recv_frame_size=8000,clock_source=external,sync=external
[INFO] [UHD RF] RF UHD Generic instance constructed
[INFO] [X300] X300 initialization sequence...
[INFO] [X300] Maximum frame size: 8000 bytes.
[INFO] [X300] Radio 1x clock: 184.32 MHz
[INFO] [GPS] Found an internal GPSDO: LC_XO, Firmware Rev 0.929a
[WARNING] [GPS] update_cache: Malformed GPSDO string: LC_XO, Firmware Rev 0.929a
[INFO] [0/DmaFIFO_0] Initializing block control (NOC ID: 0xF1F0D00000000000)
[INFO] [0/DmaFIFO_0] BIST passed (Throughput: 1318 MB/s)
[INFO] [0/DmaFIFO_0] BIST passed (Throughput: 1301 MB/s)
[INFO] [0/Radio_0] Initializing block control (NOC ID: 0x12AD100000000001)
[INFO] [0/Radio_1] Initializing block control (NOC ID: 0x12AD100000000001)
[INFO] [0/DDC_0] Initializing block control (NOC ID: 0xDDC0000000000000)
[INFO] [0/DDC_1] Initializing block control (NOC ID: 0xDDC0000000000000)
[INFO] [0/DUC_0] Initializing block control (NOC ID: 0xD0C0000000000000)
[INFO] [0/DUC_1] Initializing block control (NOC ID: 0xD0C0000000000000)
Waiting PHY to initialize ... done!
Attaching UE...
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=9294
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=9454
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=9614
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=9774
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=9934
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=10094
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=14
q
Stopping ..
---  exiting  ---

**Comment by oliverxsch** [comment]
> could you try the following change for the UE config:
> 
> ```
> device_args = type=x300,addr=192.168.43.2,master_clock_rate=184.32e6,send_frame_size=8000,recv_frame_size=8000,clock_source=external,sync=external
> ```
> 
> ie `clock` --> `clock_source` and post the console output here

Thanks for your answer, i will be able to try this on tuesday.

**Reply by pgawlowicz** [reply]
ok, then you need to go back to `clock` and probably tune tx/rx gains on both sides. Probably also time_adv_nsamples in UE ( and maybe time_alignment_calibration in gnb).

**Reply by pgawlowicz** [reply]
Hi @oliverxsch, any progress on this issue?
